libs/core/src/lib/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,268 @@
+
+    /** Event fired when the dragged file enters the component boundaries. */
+    @Output()
+    readonly onDragEnter: EventEmitter<void> = new EventEmitter<void>();
+
+    /** Event fired when the dragged file exits the component boundaries. */
+    @Output()
+    readonly onDragLeave: EventEmitter<void> = new EventEmitter<void>();
+
+    constructor(private changeDetRef: ChangeDetectorRef) { }
+
+    /** @hidden */
+    onChange: Function = () => { };
+
+    /** @hidden */
+    onTouched: Function = () => { };
+
+
+
+    /** @hidden */
+    registerOnChange(fn: any): void {
+        this.onChange = fn;
+    }
+
+    /** @hidden */
+    registerOnTouched(fn: any): void {
+        this.onTouched = fn;
+    }
+
+    /** @hidden */
+    setDisabledState(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+        this.changeDetRef.detectChanges();
+    }
+
+    /** @hidden */
+    writeValue(files: File[]): void {
+        // not needed - should be handled by user.
+    }
+
+    /** @hidden */
+    selectHandler(event: File[]): void {
+
+        this.validateFiles(event);
+        if (this.validFiles.length > 0) {
+            this.setInputValue(this.validFiles);
+            this.onChange(this.validFiles);
+            this.selectedFilesChanged.emit(this.validFiles);
+        }
+        if (this.invalidFiles.length > 0) {
+            this.selectedInvalidFiles.emit(this.invalidFiles);
+        } else {
+            this.setInputValue(event);
+            this.onChange(event);
+            this.selectedFilesChanged.emit(event);
+        }
+
+    }
+
+    validateFiles(event: File[]): void {
+        if (this.fileLimit && event.length > this.fileLimit) {
+            throw new Error('FileLimitError - Selected files count is more than specified limit ');
+        }
+        if (this.maxFileSize && this.minFileSize) {
+            const maxsize = this.parseFileSize(this.maxFileSize);
+            const minsize = this.parseFileSize(this.minFileSize);
+            event.forEach((file) => {
+                if (file.size < maxsize && file.size > minsize) {
+                    this.validFiles.push(file);
+                } else {
+                    this.invalidFiles.push(file);
+                }
+            });
+        } else if (this.maxFileSize) {
+            const maxsize = this.parseFileSize(this.maxFileSize);
+            event.forEach((file) => {
+                if (file.size < maxsize) {
+                    this.validFiles.push(file);
+                } else {
+                    this.invalidFiles.push(file);
+                }
+            });
+        } else if (this.minFileSize) {
+            const minsize = this.parseFileSize(this.minFileSize);
+            event.forEach((file) => {
+                if (file.size > minsize) {
+                    this.validFiles.push(file);
+                } else {
+                    this.invalidFiles.push(file);
+                }
+            });
+        }
```

Comment:
```suggestion

            const checkSize = (fileSize: number): boolean => {
                if (this.maxFileSize &&  fileSize > maxSize) { return false; }
                if (this.maxFileSize &&  fileSize < minSize) { return false; }
                return true;
            }
            
            this.validFiles = this.files.filter(file => checkSize(file.size));
            this.invalidFiles = this.files.filter(file => !checkSize(file.size));
```

Reply:
I think ti will be much cleaner than 3 * forEach. From performance perspective is much worse, but I don't think if anyone will apply more than 100 files.

Reply:
modfied.